src/MigrationTools.Clients.AzureDevops.ObjectModel/FieldMaps/MultiValueConditionalMap.cs

```diff
@@ -51,7 +51,7 @@ private bool fieldsExist(Dictionary<string, string> fieldsAndValues, WorkItem wo
 
         private void fieldsUpdate(Dictionary<string, string> fieldAndValues, WorkItem workitem)
         {
-            foreach (string field in fieldAndValues.Keys)
+            foreach (string field in fieldAndValues?.Keys)
```

Comment:
That will raise an exception in case `fieldAndValues` is null.
It should be: `foreach (string field in fieldAndValues?.Keys ?? Enumerable.Empty<string>())`

Same on all other foreach instances of course. The logging part is ok where null will be treated as `string.Empty` :)

Reply:
I changed the logic @gereon77 :) thanks